packages/core/src/app.ts

```diff
@@ -1545,8 +1578,14 @@ export class App {
                             h = new URL(field.value).host;
                         } catch (e) {}
 
-                        // If host doesn't match exactly, try with/without "www."
-                        return h === host || (host.startsWith("www.") ? host.slice(4) === h : "www." + host === h);
+                        const domainOptions = this.state.index.splitHostBySubdomains(host);
+
+                        // If host doesn't match exactly, try with/without "www." and all subdomain variations
+                        return (
+                            h === host ||
+                            (host.startsWith("www.") ? host.slice(4) === h : "www." + host === h) ||
+                            domainOptions.some((domain) => domain === h)
+                        );
```

Comment:
With the suggested change from above (making `splitHostBySubdomains` function return **all** variants), this could be simplified to:

```ts
return this.state.index.getHostnameVariants(host).includes(h)
```